common/scripted_effects/wc_professions_effects.txt

```diff
@@ -0,0 +1,158 @@
+		target = flag:$RECIPE$
+	}
+	custom_tooltip = discover_recipe_tt
+}
+
+save_profession_effect = {
+	if = {
+		limit = { $PROFESSION$ = flag:Mining }
+		add_trait = profession_mining
+		save_scope_value_as = { name = new_profession value = flag:profession_mining }
+		setup_profession_vars_effect = {
+			PROFESSION = flag:mining
+		}
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Herbalism }
+		add_trait = profession_herbalism
+		save_scope_value_as = { name = new_profession value = flag:profession_herbalism }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Skinning }
+		add_trait = profession_skinning
+		save_scope_value_as = { name = new_profession value = flag:profession_skinning }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Blacksmithing }
+		add_trait = profession_blacksmithing
+		save_scope_value_as = { name = new_profession value = flag:profession_blacksmithing }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Alchemy }
+		add_trait = profession_alchemy
+		save_scope_value_as = { name = new_profession value = flag:profession_alchemy }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Enchanting }
+		add_trait = profession_enchanting
+		save_scope_value_as = { name = new_profession value = flag:profession_enchanting }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Engineering }
+		add_trait = profession_engineering
+		save_scope_value_as = { name = new_profession value = flag:profession_engineering }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Inscription }
+		add_trait = profession_inscription
+		save_scope_value_as = { name = new_profession value = flag:profession_inscription }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Jewelcrafting }
+		add_trait = profession_jewelcrafting
+		save_scope_value_as = { name = new_profession value = flag:profession_jewelcrafting }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Cooking }
+		add_trait = profession_cooking
+		save_scope_value_as = { name = new_profession value = flag:profession_cooking }
+	}
+	if = {
+		limit = { $PROFESSION$ = flag:Tailoring }
+		add_trait = profession_tailoring
+		save_scope_value_as = { name = new_profession value = flag:profession_tailoring }
+	}
+	if = {
+		limit = {
+			var:profession_1 = flag:none
+			var:profession_2 = flag:none
+		}
+		set_variable = {
+			name = profession_1
+			value = scope:new_profession
+		}
+		set_variable = {
+			name = profession_1_cd
+			value = yes
+			days = 365
+		}
+	}
+	else_if = {
+		limit = {
+			NOT = { var:profession_1 = flag:none }
+			var:profession_2 = flag:none
+		}
+		set_variable = {
+			name = profession_2
+			value = scope:new_profession
+		}
+		set_variable = {
+			name = profession_2_cd
+			value = yes
+			days = 365
+		}
+	}
+}
+
+set_cd_effect = {
+	if = {
+		limit = { has_character_flag = profession_1_cd }
+		set_variable = {
+			name = profession_1_oncd
+			value = yes
+		}
+	}
+	if = {
+		limit = { has_character_flag = profession_2_cd }
+		set_variable = {
+			name = profession_2_oncd
+			value = yes
+		}
```

Comment:
can we change these flags from cd / oncd to `cooldown`/`on_cooldown` it's just better to fully write things out

Reply:
i am removing gthis